src/py42/clients/alertrules/cloud_share.py

```diff
@@ -0,0 +1,26 @@
+import json
+
+from py42.clients import BaseClient
+
+
+class CloudShareClient(BaseClient):
+
+    _version = u"v1"
+    _resource = u"query-cloud-share-permissions-rule"
+    _api_prefix = u"/svc/api/{0}/Rules/{1}".format(_version, _resource)
+
+    def __init__(self, session, tenant_id):
+        super(CloudShareClient, self).__init__(session)
+        self._tenant_id = tenant_id
+
+    def get(self, rule_id):
+        """Fetch cloud share alert rule by rule id.
+
+        Args:
+            rule_id (str): Observer rule Id of a rule to be fetched.
+
+        Returns
+            :class:`py42.response.Py42Response`
+        """
+        data = {"tenantId": self._tenant_id, "ruleIds": [rule_id]}
```

Comment:
The three specific rule clients all seem really similar and could share code. Do we want to make a base `RuleClient` that handles creating the data and making the request and takes a URL?

Reply:
this is the case right now, but the type of data required to update each type of rule is very different. I'd rather keep them separate.

Reply:
This idea of similar code just allows me to share a thought I was having, all we do as business logic here is

-  map the method parameters to request fields of the API, 

- define uri

- define http method 

so even that can be abstracted out to a configuration file and a function,  this would reduce the verbosity, and only focus would remain to design the feature layout.

Obviously this would be a wrapper around a wrapper and as it adds code complexity, not convinced enough to prefer such approach.

Reply:
that's an interesting thought for py42 as a whole! probably out of scope for this PR though